### DISCUSSION TITLE
QNAP NAS TS-431P2 cannot connect via VScode ssh

### DISCUSSION THREAD

**Discussion by Horia62** [microsoft/vscode-discussions#626] [Extension Development QnA] [answered]
Hi to all,
I need the functionality to be be able to connect to my QNAP NAS from win10 via ssh using VScode Remote-ssh.
In the process of running the ssh from VScode it seems that some command are not recognized by the NAS so the ssh connection is not established:
[09:31:39.387] Showing password prompt
[09:31:45.939] Got password response
[09:31:45.939] "install" wrote data to terminal: "***********"
[09:31:45.953] >
[09:31:46.059] > 3a63c10769b3: running
[09:31:46.076] > grep: invalid option -- 'a'
> BusyBox v1.01 (2023.04.20-19:11+0000) multi-call binary
>
> Usage: grep [-ihHnqvs] PATTERN [FILEs...]
>
> Search for PATTERN in each FILE or standard input.
>
> Options:
> -H prefix output lines with filename where match was found
> -h suppress the prefixing filename on output
> -i ignore case distinctions
> -l list names of files that match
> -n print line number with output lines
> -q be quiet. Returns 0 if result was found, 1 otherwise
> -v select non-matching lines
> -s suppress file open/read error messages
>
> grep: invalid option -- 'a'
> BusyBox v1.01 (2023.04.20-19:11+0000) multi-call binary

This is the only message that I get looking on output and the:
Setting up SSH Host xxx.xxx.xxx.xxx: ([details](command:opensshremotes.showDetails)) Initializing VS Code ServerAll the credentials are correct
is never performed.

Does anyone had this issue? Is there a solution for this? Thank you.

**Comment by gjsjohnmurray** [comment, marked as answer]
This forum is intended for developers who are creating extensions for VS Code. Your issue is better suited to the main VS Code repo at https://github.com/microsoft/vscode/issues, but even there you might get advised to ask at https://stackoverflow.com/questions/tagged/visual-studio-code

**Reply by Horia62** [reply]
Sorry for this. As my first post I guess that here was the correct place. I will give it a try on the mentioned links.
My apologies, and thank you!
